libsnark/gadgetlib1/gadgets/hashes/anemoi/tests/test_anemoi_gadget.cpp

```diff
@@ -0,0 +1,221 @@
+
+template<typename ppT> void test_flystel_E_power_five_gadget(const size_t n)
+{
+    using FieldT = libff::Fr<ppT>;
+    printf("testing flystel_E_power_five_gadget on all %zu bit strings\n", n);
+
+    protoboard<FieldT> pb;
+    pb_variable<FieldT> x;
+    pb_variable<FieldT> y;
+
+    // input
+    x.allocate(pb, "x");
+    // output
+    y.allocate(pb, "y");
+
+    // create gadget
+    flystel_E_power_five_gadget<FieldT> d(pb, x, y, "d");
+    // generate contraints
+    d.generate_r1cs_constraints();
+    // set input value
+    pb.val(x) = 2;
+    // generate witness for the given input
+    d.generate_r1cs_witness();
+
+    // the expected output is 32 for input 2
+    ASSERT_EQ(pb.val(y), 32);
+    ASSERT_TRUE(pb.is_satisfied());
+
+    libff::print_time("flystel_E_power_five_gadget tests successful");
+}
+
+template<typename ppT> void test_flystel_E_root_five_gadget(const size_t n)
+{
+    using FieldT = libff::Fr<ppT>;
+    printf("testing flystel_E_root_five_gadget on all %zu bit strings\n", n);
+
+    protoboard<FieldT> pb;
+    pb_variable<FieldT> x;
+    pb_variable<FieldT> y;
+
+    // input
+    x.allocate(pb, "x");
+    // output
+    y.allocate(pb, "y");
+
+    // create gadget
+    flystel_E_root_five_gadget<FieldT> d(pb, x, y, "d");
+    // generate contraints
+    d.generate_r1cs_constraints();
+    // set input value
+    pb.val(x) = 22;
+    // generate witness for the given input
+    d.generate_r1cs_witness();
+
+    // computed using Sage
+    FieldT y_expected = FieldT("10357913779704000956629425810748166374506105653"
+                               "828973721142406533896278368512");
+
+    // the expected output is 32 for input 2
+    ASSERT_EQ(pb.val(y), y_expected);
+    ASSERT_TRUE(pb.is_satisfied());
+
+    libff::print_time("flystel_E_root_five_gadget tests successful");
+}
+
+template<typename ppT> void test_flystel_prime_field_gadget(const size_t n)
+{
+    using FieldT = libff::Fr<ppT>;
+    printf("testing flystel_prime_field_gadget on all %zu bit strings\n", n);
+
+    protoboard<FieldT> pb;
+
+    pb_variable<FieldT> x0 = pb_variable_allocate(pb, "x0");
+    pb_variable<FieldT> x1 = pb_variable_allocate(pb, "x1");
+    pb_variable<FieldT> y0 = pb_variable_allocate(pb, "y0");
+    pb_variable<FieldT> y1 = pb_variable_allocate(pb, "y1");
+
+    flystel_prime_field_gadget<
+        FieldT,
+        FLYSTEL_MULTIPLICATIVE_SUBGROUP_GENERATOR>
+        d(pb, x0, x1, y0, y1, "flystel");
+
+    // generate constraints
+    d.generate_r1cs_constraints();
+
+    pb.val(x0) = FieldT(55);
+    pb.val(x1) = FieldT(3);
+
+    // generate witness for the given input
+    d.generate_r1cs_witness();
+
+    FieldT y0_expect = FieldT(34);
+    FieldT y1_expect = FieldT(1);
+
+    ASSERT_EQ(y0_expect, pb.val(y0));
+    ASSERT_EQ(y1_expect, pb.val(y1));
+    ASSERT_TRUE(pb.is_satisfied());
+
+    //    test_pb_verify_circuit<ppT>(pb);
+
+    libff::print_time("flystel_prime_field_gadget tests successful");
+}
+
+int main()
```

Comment:
switch to gtest framework

Reply:
Will be addressed after `anemoi-hash-r1cs-parametrization` is merged into `anemoi-hash-r1cs` .

Reply:
#77

Reply:
Addressed.